pdr_backend/analytics/predictoor_dashboard/dash_components/plots.py

```diff
@@ -0,0 +1,151 @@
+from typing import Union, List, Tuple, Optional
+import plotly.graph_objects as go
+from enforce_typing import enforce_types
+from pdr_backend.util.time_types import UnixTimeS
+
+
+@enforce_types
+def create_scatter(
+    name: str, x_data: List[float], y_data: List[int], mode="lines"
+) -> go.Scatter:
+    """Create a scatter plot with the given data.
+    Args:
+        name (str): Name of the trace.
+        x_data (List): List of x values.
+        y_data (List): List of y values.
+        mode (str): Plot mode. Default is 'lines'.
+    Returns:
+        go.Scatter: Scatter plot trace.
+    """
+    return go.Scatter(x=x_data, y=y_data, mode=mode, name=name)
+
+
+@enforce_types
+def create_bar(name: str, x_data: List[float], y_data: List[int]) -> go.Bar:
+    """Create a bar plot with the given data.
+    Args:
+        name (str): Name of the trace.
+        x_data (list): List of x values.
+        y_data (list): List of y values.
+    Returns:
+        go.Bar: Bar plot trace.
+    """
+    return go.Bar(x=x_data, y=y_data, name=name, width=1)
+
+
+@enforce_types
+def process_payouts(payouts: List[dict], predictor: str, feed: str) -> tuple:
+    """
+    Process payouts data for a given predictor and feed.
+    Args:
+        payouts (list): List of payouts data.
+        predictor (str): Predictor address.
+        feed (str): Feed contract address.
+    Returns:
+        tuple: Tuple of slots, accuracies, profits, and stakes.
+    """
+    slots, accuracies, profits, stakes = [], [], [], []
+    profit = predictions = correct_predictions = 0
+
+    for p in payouts:
+        if predictor in p["ID"] and feed in p["ID"]:
```

Comment:
let's do an "if inversion".

if the opposite of this is true, add a continue statement
unindent the rest of the block